dbms/src/Flash/Mpp/ExchangeReceiver.h

```diff
@@ -213,6 +214,13 @@ class ExchangeReceiverBase
 
     bool collected = false;
     int thread_count = 0;
+    uint64_t fine_grained_shuffle_stream_count;
+
+    /// True when:
+    ///   1. we are in disaggregated tiflash mode
+    ///   2. and this receiver is receiving data from tiflash_storage node.
+    /// If it's true, when this ExchangeReciever is canceled, need to cancel MPPTasks in tiflash_storage node.
+    bool is_tiflash_storage_receiver;
```

Comment:
Can we think of a more explicit way to cancel MPPTasks on storage node, rather than cascade the cancel call from compute to storage.

My major concern is that this cascade cancel won't be needed if we replace current `StorageDisaggregated` with a decent one in the future. When that happens, I would like to see the "canceling MPPTasks on storage node" thing gone with the old `StorageDisaggregated`, without need for removing any code in `ExchangeReceiver` itself.

Reply:
`StorageDisaggregated` works in compilation phase, `cancel` happens in compilation or execution phase. Guess it's little difficult to combine `cancel` logic into `StorageDisaggregated`.

But I can encapsulate `cancel` logic into one function, so later we can remove it very convenient.
(PS: when the real `StorageDisaggregated` is ready, we still need some way to read data from tiflash_storage node, this process may still need to cancel xxxTask on storage node)

Reply:
@windtalker Could you please suggest a place to put this explicit cancel logic of MPPTasks in tiflash storage?